src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/VirtualProjectXmlProviderTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.LanguageServer.FileBasedPrograms;
+using Microsoft.Extensions.Logging;
+using Roslyn.LanguageServer.Protocol;
+using Roslyn.Test.Utilities;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests;
+
+/// <summary>
+/// Goal of these tests:
+/// - Ensure that the various request/response forms work as expected in basic scenarios.
+/// - Ensure that various properties on the response are populated in a reasonable way.
+/// Non-goals:
+/// - Thorough behavioral testing.
+/// - Testing of more intricate behaviors which are subject to change.
+/// </summary>
+public sealed class VirtualProjectXmlProviderTests : AbstractLanguageServerHostTests
+{
+    public VirtualProjectXmlProviderTests(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+    {
+    }
+
+    private async Task<VirtualProjectXmlProvider> GetProjectXmlProviderAsync()
+    {
+        var (exportProvider, assemblyLoader) = await LanguageServerTestComposition.CreateExportProviderAsync(
+            LoggerFactory, includeDevKitComponents: false, MefCacheDirectory.Path, extensionPaths: null);
+
+        return exportProvider.GetExportedValue<VirtualProjectXmlProvider>();
+    }
+
+    [Fact]
+    public async Task GetProjectXml_FileBasedProgram_01()
+    {
+        var projectProvider = await GetProjectXmlProviderAsync();
+
+        var tempDir = TempRoot.CreateDirectory();
+        var appFile = tempDir.CreateFile("app.cs");
+        await appFile.WriteAllTextAsync("""
+            Console.WriteLine("Hello, world!");
+            """);
+
+        var globalJsonFile = tempDir.CreateFile("global.json");
+        // TODO: these tests should be conditioned on availability of an SDK which supports run-api
+        await globalJsonFile.WriteAllTextAsync("""
+            {
+              "sdk": {
+                "version": "10.0.100-preview.5.25265.12"
```

Comment:
these tests should fail in CI (and look like they are).  We would need to figure out some way of ensuring the preview SDK is available in helix machines.